src/testmenu.cpp

```diff
@@ -0,0 +1,778 @@
+                }
+    #endif
+    #if ALT_STEPPER_TYPE != STEPPER_TYPE_NONE
+                else if (action == F("MoveALTAxis"))
+                {
+                    String output = String(F("Moving ALT axis by ")) + String(_secondaryDistance, 1) + String(F(" arcMins ("));
+                    output += String(ALTITUDE_STEPS_PER_ARC_MINUTE * _secondaryDistance, 0) + " steps) " + actionArg;
+                    Serial.println(output);
+                    float arcmins        = actionArg == "UP" ? _secondaryDistance : -_secondaryDistance;
+                    TestMenu::_startALT  = mount.getCurrentStepperPosition(ALTITUDE_STEPS);
+                    TestMenu::_targetALT = TestMenu::_startALT + arcmins * ALTITUDE_STEPS_PER_ARC_MINUTE;
+                    mount.moveBy(ALTITUDE_STEPS, arcmins);
+                    _internalState |= DISPLAY_ALT;
+                }
+    #endif
+                else if (action == F("ToggleTRK"))
+                {
+                    if (mount.isSlewingTRK())
+                    {
+                        mount.stopSlewing(TRACKING);
+                        Serial.println(F("Tracking stopped."));
+                    }
+                    else
+                    {
+                        mount.startSlewing(TRACKING);
+                        Serial.println(F("Tracking started."));
+                    }
+                    _currentMenu->display();
+                }
+                else if (action == F("FactoryReset"))
+                {
+                    mount.clearConfiguration();
+                    Serial.println(F("Mount reset, EEPROM erased."));
+                    _currentMenu->display();
+                }
+                else if (action = F("PassthroughCmd"))
+                {
+                    Serial.print(F("Enter LX200-OAT command to send to mount: "));
+                    TestMenu::setMenuState(testMenuState_t::WAITING_ON_COMMAND);
+                }
+            }
+            if (TestMenu::getMenuState() == testMenuState_t::CLEAR)
+            {
+                _currentMenu->display();
+            }
+            return;
+        }
+    }
+    Serial.println(F("Invalid key pressed."));
+
+    _currentMenu->display();
+}
+
+void TestMenu::displayStepperPos() const
+{
+    String statusRaDec;
+    statusRaDec = F("  RA: ");
+    statusRaDec += rightJustify(String(mount.getCurrentStepperPosition(RA_STEPS)), 8);
+    statusRaDec += mount.isAxisRunning(RA_STEPS) ? "^" : " ";
+    statusRaDec += F("   ALT: ");
+    #if (ALT_STEPPER_TYPE != STEPPER_TYPE_NONE)
+    String altSteps = "n/a";
+    #else
+    String altSteps = String(mount.getCurrentStepperPosition(ALTITUDE_STEPS));
+    #endif
+    statusRaDec += rightJustify(altSteps, 8);
+    statusRaDec += mount.isAxisRunning(ALTITUDE_STEPS) ? "^" : " ";
+    statusRaDec += F("   TRK: ");
+    statusRaDec += rightJustify(String(mount.getCurrentStepperPosition(TRACKING)), 8);
+    statusRaDec += mount.isSlewingTRK() ? "^" : " ";
```

Comment:
This would be a lot more efficient by doing something like this:
```C++
    char statusRaDec[100];
    snprintf(statusRaDec,
             ARRAY_SIZE(a),
             "  RA: \n"
             "        %lu\n"
             "%c\n"
             "   ALT: ",
             ...etc
             mount.getCurrentStepperPosition(RA_STEPS),
             mount.isAxisRunning(RA_STEPS) ? '^' : ' ');
```
(Arduino `String` uses a lot of dynamic allocation, especially when adding strings like this)
I also think it's more understandable on what the strings look like.

Reply:
This doesn't give the same result, though. Original code keeps the columns lined up, new code does not. Does snprintf support justifications?

Reply:
Let me see if Arduino lib supports that....

Reply:
OK, yeah, that works. It uses a lot more stack now, though, 64 bytes for the buffer and 2 x 42 for the format strings.

Reply:
It looks like you can use some variant of `PSTR()` or `snprintf_P` to put the format string into program memory: https://forum.arduino.cc/t/resolved-without-hacks-how-i-used-f-flash-memory-strings-in-snprintf-etc-for-arduino-nano/891927/4

Reply:
Oh, nice! Yeah, that saved a bunch of memory.

Reply:
Yes! Good catch.